doc/Bucket_Testing.md

```diff
@@ -0,0 +1,152 @@
+# BucketTesting
+
+The bucket testing feature allows us to distribute visitors in "test groups" (buckets) and 
+make the Fundraising Application behave differently, depending on the bucket a visitor is in. 
+It also logs the selected buckets of a visitor when an application event occurs 
+(e.g. visitors donates, visitor applies for membership, etc) 
+
+The goal of the bucket testing is to measure which application behavior change leads to an 
+increase in one of our metrics (number of donations, number of non-anonymous donations, 
+number of recurring donations, etc.).
+
+We chose the name "Bucket Testing" instead of the more commonly know name "A/B testing" 
+because the system is not limited to two buckets per test.
+
+We designed the system to run bucket tests without URL parameters from banners, 
+but we rarely used that functionality.
+
+This is a reference documentation for developers, 
+see [How to set up an A/B test](HOWTO_Create_an_a_b_test.md) 
+for practical steps for setting up campaigns
+
+## Domain
+
+### Bucket
+Each bucket has a *name* and a reference to the *campaign* that it belongs to.
+The name must be unique for that campaign.
+
+Each bucket can generate a **Bucket ID**, consisting of the prefix `campaign`,
+the campaign name and the bucket name, separated by dots.
+
+Class: `WMDE\Fundraising\Frontend\BucketTesting\Domain\Model\Bucket`
+
+### Campaigns
+Campaigns are a collection of *Buckets*. One of these buckets is the **Default Bucket** - 
```

Comment:
```suggestion
Campaigns can have 2 or more *Buckets*. One of these buckets is the **Default Bucket** - 
```
very nitpicky:
semantically it makes more sense to me that the campaign can exist without the buckets => the set of buckets does not define the campaign